File: JavaCoding/Array/ArrayLatest/mostAsked/MinimumAbsolutedifference.java
package JavaCoding.Array.ArrayLatest.mostAsked;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class MinimumAbsolutedifference {

    public static void main(String[] args) {

        int [] arr ={4,1,2,3};

        int len = arr.length-1;
        int min = Integer.MAX_VALUE;
        List<List<Integer>> res = new ArrayList<>();
        Arrays.sort(arr);

        for(int i = 0;i<len;i++){

            int diff = arr[i+1]-arr[i];
            min = Math.min(diff,min);

        }

        for(int i = 0;i<len;i++){

            int diff = arr[i+1]-arr[i];
            if(min==diff)
             res.add(Arrays.asList(arr[i],arr[i+1]));

        }

       

        System.out.println(res);
        
    }


    
}


 /*
        // Single pass: track min and collect pairs
        
        for (int i = 0; i < arr.length - 1; i++) {
            int diff = arr[i + 1] - arr[i];

            if (diff < min) {
                min = diff;
                list.clear(); // New smaller min — reset the list
                list.add(Arrays.asList(arr[i], arr[i + 1]));
            } else if (diff == min) {
                list.add(Arrays.asList(arr[i], arr[i + 1]));
            }
        }
       */

File: Javascript/MostAsked/MinimumAbsoluteDifference.js
let arr = [4,2,1,3];

let len = arr.length-1;
let min = Infinity;

const result = [];

arr.sort((a,b)=>a-b);

for(let i=0;i<len;i++){

     let diff = arr[i+1]-arr[i];
     min = Math.min(diff,min);
}

for(let i=0;i<len;i++){

    let diff = arr[i+1]-arr[i];
    if(min===diff){
        result.push(arr[i],arr[i+1]);
}
    
}

console.log(result);

